README.md

```diff
@@ -114,43 +163,276 @@ encoding as JSON. Since Python is a dynamic language, when decoding via
+        self._complete = asyncio.Event()
+
+    @workflow.run
+    async def run(self, name: str) -> str:
+        self._greeting_info.name = name
+        while True:
+            # Store greeting
+            self._current_greeting = await workflow.execute_activity(
+                create_greeting_activity,
+                self._greeting_info,
+                start_to_close_timeout=timedelta(seconds=5),
+            )
+            workflow.logger.debug("Greeting set to %s", self._current_greeting)
+            
+            # Wait for salutation update or complete signal (this can be
+            # cancelled)
+            await asyncio.wait(
+                [self._greeting_info_update.wait(), self._complete.wait()],
+                return_when=asyncio.FIRST_COMPLETED,
+            )
+            if self._greeting_info_update.is_set():
+                self._greeting_info_update.clear()
+            elif self._complete.is_set():
+                return self._current_greeting
+
+    @workflow.signal
+    async def update_salutation(self, salutation: str) -> None:
+        self._greeting_info.salutation = salutation
+        self._greeting_info_update.set()
+
+    @workflow.signal
+    async def complete_with_greeting(self) -> None:
+        self._complete.set()
+
+    @workflow.query
+    async def current_greeting(self) -> str:
+        return self._current_greeting
 
 @activity.defn
-async def say_hello_activity(name: str) -> str:
-    return f"Hello, {name}!"
+async def create_greeting_activity(info: GreetingInfo) -> str:
+    return f"{info.salutation}, {info.name}!"
+```
 
+Some things to note about the above code:
 
-async def main(stop_event: asyncio.Event):
-  # Create client connected to server at the given address
-  client = await Client.connect("http://localhost:7233", namespace="my-namespace")
+* This workflow continually updates the queryable current greeting when signalled and can complete with the greeting on
+  a different signal
+* Workflows are always classes and must have a single `@workflow.run` which is an `async def` function
+* Workflow code must be deterministic. This means no threading, no randomness, no external calls to processes, no
+  network IO, and no global state mutation. All code must run in the implicit `asyncio` event loop and be deterministic.
+* `@activity.defn` is explained in a later section. For normal simple string concatenation, this would just be done in
+  the workflow. The activity is for demonstration purposes only.
+* `workflow.execute_activity(create_greeting_activity, ...` is actually a typed signature, and MyPy will fail if the
+  `self._greeting_info` parameter is not a `GreetingInfo`
+
+Here are the decorators that can be applied:
+
+* `@workflow.defn` - Defines a workflow class
+  * Must be defined on the class given to the worker (ignored if present on a base class)
+  * Can have a `name` param to customize the workflow name, otherwise it defaults to the unqualified class name
```

Comment:
```suggestion
  * Can have a `name` param to customize the workflow name (`@workflow.defn(name="my workflow")`), otherwise it defaults to the unqualified class name
```

Reply:
You're overthinking intentionally sparse documentation. There are dozens of bullet points I could put snippets showing what I mean by "parameters" or whatever else I reference.